File: typegate/tests/metagen/typegraphs/identities/py/handlers.py
from .handlers_types import (
    Composites,
    typed_cycles,
    Primitives,
    typed_composites,
    Cycles1,
    typed_primitives,
    SimpleCycles1,
    typed_simple_cycles,
)
from . import handlers_types as types


@typed_primitives
def primitives(inp: types.PrimitivesArgs) -> Primitives:
    return inp.data


@typed_composites
def composites(inp: types.CompositesArgs) -> Composites:
    return inp.data


@typed_cycles
def cycles(inp: types.Cycles1Args) -> Cycles1:
    return inp.data


@typed_simple_cycles
def simple_cycles(inp: types.SimpleCycles1Args) -> SimpleCycles1:
    return inp.data

File: typegate/tests/metagen/typegraphs/identities/ts/handlers.ts
// Copyright Metatype OÜ, licensed under the Elastic License 2.0.
// SPDX-License-Identifier: Elastic-2.0

import type {
  Composites,
  Cycles1,
  Primitives,
  SimpleCycles1,
  TsCompositesHandler,
  TsCyclesHandler,
  TsPrimitivesHandler,
  TsSimpleCyclesHandler,
} from "./mdk.ts";

export const primitives: TsPrimitivesHandler = (inp, _ctx, _tg) => {
  const out: Primitives = {
    ...inp.data,
  };
  return out;
};

export const cycles: TsCyclesHandler = (inp, _ctx, _tg) => {
  const out: Cycles1 = {
    ...inp.data,
  };
  return out;
};

export const composites: TsCompositesHandler = (inp, _ctx, _tg) => {
  const out: Composites = {
    ...inp.data,
  };
  return out;
};

export const simple_cycles: TsSimpleCyclesHandler = (inp, _ctx, _tg) => {
  const out: SimpleCycles1 = {
    ...inp.data,
  };
  return out;
};
